feat: add title to interface to inline requestBody and responses

simllll opened the pull request:
### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/lukeautry/tsoa/tree/master/docs/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/lukeautry/tsoa/pulls) for the same update/change?
- [ ] Have you written unit tests?
- [ ] Have you written unit tests that cover the negative cases (i.e.: if bad data is submitted, does the library respond properly)?
- [x] This PR is associated with an existing issue?

**Closing issues**

closes #1243

### If this is a new feature submission:

- [ ] Has the issue had a maintainer respond to the issue and clarify that the feature is something that aligns with the [goals](https://github.com/lukeautry/tsoa#goal) and [philosophy](https://github.com/lukeautry/tsoa#philosophy) of the project?

This feature allows more consistent client generation as the body and response objects where without a name before and therefore client genreation lead to "inline objects" naming without a qualified name.

before: 
![image](https://user-images.githubusercontent.com/5757263/165069052-a74fe345-feb2-4413-8097-f457f2f9e0e0.png)
after: 
![image](https://user-images.githubusercontent.com/5757263/165069128-b7fe4a43-e8dd-4e67-b495-64ccfc177fb9.png)


**Potential Problems With The Approach**

as this is an additional optional field, I think there are no potential problems

simllll commented:
Offtopic: but locally the esm tests are failing when running npm run test

```
> tsoa-tests-esm@4.0.0-beta.0 pretest
> node --experimental-loader=ts-node/esm prepare.ts

(node:793294) ExperimentalWarning: --experimental-loader is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
TypeError: Unknown file extension ".ts" for /home/simon/WebstormProjects/tsoa/tests/esm/prepare.ts

```

is there anything special I need to consider?

simllll commented:
@WoH any feedback on that? :)

WoH commented:
I've looked at the PR, there are 2 things that I wanna get some feedback on: First I want to find out what all the unexpected ways are this can break stuff. Secondly, I'm not sure the location for the title determination is ideal, maybe we can do that earlier and attach it to the type?

simllll commented:
Thanks for coming back to me!

Regarding the first point: What do you think if we just hide this feature behind a flag, and only set the title if this flag is set? This would prevent all unknown side effects!

And about the second one: I was thinking and playing around about the place where to add this title prop, and it turned out this was the place with the least modifications and single places for the title definition. Adding it earlier in the type itself, made it a bit unclear (in my opinion) where the title is coming from and why, because an object can have multiple places where it can occur it also bloats the code quite a lot to have all the cases handled. Or did you mean something else?

simllll commented:
@WoH I added useTitleTagsForInlineObjects to the config and hide the feature behind this flag.

WoH commented:
Code looks good, but can you add a test that uses this setting?

simllll commented:
I've added two test cases, but I stumpled upon something else, it seems the schema definition is not 100% correct.
I've fixed it swagger.ts on line 220, but I think it could also be wrong at line 215.

simllll commented on packages/runtime/src/swagger/swagger.ts:
```diff
@@ -217,7 +217,7 @@ export namespace Swagger {
   export interface Response3 {
     description: string;
-    content?: { [name: string]: Schema & Example };
+    content?: { [name: string]: { schema?: Schema } & Example };
```
can schema really be optional? It's as optional everywehre else, therefore I did it here too.. but not sure if needed

simllll commented on packages/runtime/src/swagger/swagger.ts:
```diff
@@ -217,7 +217,7 @@ export namespace Swagger {
   export interface Response3 {
     description: string;
-    content?: { [name: string]: Schema & Example };
+    content?: { [name: string]: Schema3 & { schema?: Schema } & Example };
```
can schema really be optional? It's as optional everywehre else, therefore I did it here too.. but not sure if needed

WoH replied:
Maybe it is simply related to partial responses being built?
I don't really know an example where there wouldn't be a schema

simllll replied:
I've addressed this in a second PR https://github.com/lukeautry/tsoa/pull/1253 
schema is not optional there anymore, as there is not situation where this can be the case

simllll commented:
Any plan when this will land in the master / release? Really looking forward to this ...
